docs/docs-content/clusters/public-cloud/azure/required-permissions.md

```diff
@@ -309,659 +258,1312 @@ various scenarios.
+       "Microsoft.ContainerService/managedClusters/trustedAccessRoleBindings/delete",
+       "Microsoft.ContainerService/managedClusters/trustedAccessRoleBindings/read",
+       "Microsoft.ContainerService/managedClusters/trustedAccessRoleBindings/write",
+       "Microsoft.ContainerService/managedClusters/ui/read",
+       "Microsoft.ContainerService/managedClusters/unpinManagedCluster/action",
+       "Microsoft.ContainerService/managedClusters/upgradeProfiles/read",
+       "Microsoft.ContainerService/managedClusters/users/impersonate/action",
+       "Microsoft.ContainerService/managedClusters/version/read",
+       "Microsoft.ContainerService/managedClusters/write",
+       "Microsoft.ContainerService/managedclustersnapshots/delete",
+       "Microsoft.ContainerService/managedclustersnapshots/read",
+       "Microsoft.ContainerService/managedclustersnapshots/write",
+       "Microsoft.ContainerService/snapshots/delete",
+       "Microsoft.ContainerService/snapshots/read",
+       "Microsoft.ContainerService/snapshots/write",
+       "Microsoft.ManagedIdentity/userAssignedIdentities/assign/action",
+       "Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials/read",
+       "Microsoft.Network/virtualNetworks/delete",
+       "Microsoft.Network/virtualNetworks/read",
+       "Microsoft.Network/virtualNetworks/subnets/delete",
+       "Microsoft.Network/virtualNetworks/subnets/join/action",
+       "Microsoft.Network/virtualNetworks/subnets/read",
+       "Microsoft.Network/virtualNetworks/subnets/write",
+       "Microsoft.Network/virtualNetworks/write",
+       "Microsoft.ResourceHealth/availabilityStatuses/read",
+       "Microsoft.Resources/subscriptions/resourceGroups/read"
+     ],
+     "NotActions": [],
+     "AssignableScopes": [
+       "/subscriptions/$SUBSCRIPTION_ID"
+     ]
+   }
+   ```
+
+4. Create a role for each of the JSON files you created in the previous steps. Issue the following commands to create
+   the roles.
+
+   ```shell
+   az role definition create --role-definition @aks_dynamic_sub_role.template.json --output table
+   az role definition create --role-definition @aks_dynamic_rg_sub_role.template.json --output table
+   ```
+
+5. Export the
+   [security principal](https://learn.microsoft.com/en-us/azure/role-based-access-control/overview#security-principal)
+   object ID you want to use with Palette to a variable.
+
+   ```shell
+   export ASSIGNEE="<security_principal_object_id>"
+   ```
+
+6. Assign the roles to the service principal. Use the following commands to assign the roles.
+
+   ```json
+   az role assignment create --assignee $ASSIGNEE \
+     --role "Palette Dynamic Placement AKS Cluster Deployer (sub)" \
+     --scope "/subscriptions/$SUBSCRIPTION_ID"
+   ```
+
+   ```json
+   az role assignment create --assignee $ASSIGNEE \
+     --role "Palette Dynamic Placement AKS Cluster Deployer (rg/sub)" \
+     --scope "/subscriptions/$SUBSCRIPTION_ID"
```

Comment:
> `--scope "/subscriptions/$SUBSCRIPTION_ID"`

Same idea as above. This should be resource group scope level, unless the user has a reason to use subscription scope level.